packages/sources/layer2-sequencer-health/src/starkware.ts

```diff
@@ -0,0 +1,135 @@
+import { Logger } from '@chainlink/ea-bootstrap'
+import { DEFAULT_PRIVATE_KEY, ExtendedConfig } from './config'
+import { ec, Account, InvokeFunctionResponse, GetBlockResponse } from 'starknet'
+import { race, retry } from './network'
+
+interface StarwareState {
+  lastBlockResponse: GetBlockResponse | null
+  lastUpdated: number
+  isSequencerHealthy: boolean
+}
+
+export const sendDummyStarkwareTransaction = async (config: ExtendedConfig): Promise<void> => {
+  const starkKeyPair = ec.genKeyPair(DEFAULT_PRIVATE_KEY)
+  const starkKeyPub = ec.getStarkKey(starkKeyPair)
+  const provider = config.starkwareConfig.provider
+  const account = new Account(provider, config.starkwareConfig.dummyAccountAddress, starkKeyPair)
+
+  await race<InvokeFunctionResponse>({
+    timeout: config.timeoutLimit,
+    promise: account.execute(
+      {
+        contractAddress: config.starkwareConfig.dummyAccountAddress,
+        entrypoint: 'initialize',
+        calldata: [starkKeyPub, '0'],
+      },
+      undefined,
+      { maxFee: '0' },
+    ),
+    error: `Transaction receipt not received in ${config.timeoutLimit} milliseconds`,
+  })
+}
+
+/**
+ * The centralized Starkware sequencer is made up of two internal components.  The first is the gateway which is used to query the Sequencer state
+ * and the batcher which adds transactions to the pending block and processes the pending block to add it into the L2 chain.  The
+ * Sequencer is considered unhealthy if either of the components are down.
+ *
+ *  - The gateway is considered to be down if request to fetch pending block times out or if we get a 404
+ *  - The batcher is considered to be down if we a new block has not been produced AND no new transactions have been added to the
+ *  pending block within a given time interval.
+ * status when fetching
+ * @param networks
+ * @returns true if Sequencer is healthy
+ */
+export const checkStarkwareSequencerPendingTransactions = (): ((
+  config: ExtendedConfig,
+) => Promise<boolean>) => {
+  let recordedStarkwareState: StarwareState = {
```

Comment:
Verbose to read, suggestion:

```suggestion
  let state: StarwareState = {
```